google/provider.go

```diff
@@ -105,6 +105,7 @@ func Provider() terraform.ResourceProvider {
 			"google_container_registry_image":                 dataSourceGoogleContainerImage(),
 			"google_iam_policy":                               dataSourceGoogleIamPolicy(),
 			"google_iam_role":                                 dataSourceGoogleIamRole(),
+			"google_impersonated_credential":                  dataSourceGoogleImpersonatedCredential(),
```

Comment:
ther'es a subtle distinction between delegation and impersonation...this is really impersonation. see link [here](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16#section-1.1) from oauth2 docs about token exchange

Reply:
Touche :) Though to take a step back, we try to keep as much consistency between the the GCP api syntax and naming conventions as possible. We've done 'interpreted' naming in the past and found that it's misleading for users coming from other GCP products/docs. The reason I suggested delegated_credential is because of the language used in api documentation. I suppose it's hard to determine which pattern is happening without understanding the upstream implementation. Given that, the more implementation agnostic naming would be something derived from "serviceAccount" and "AccessToken" or "OAuthToken", so perhaps "service_account_access_token" ?

Reply:
I think its still impersonation we've got here.  Our docs [here](https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials#creating_a_limited-privilege_service_account) predominantly talk about it as impersonation (though the `delegate[]` request prameter does the same thing in chain internally).

* note, in our pyton and java client library we also called it
 - [impersonated_credential.py](https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/impersonated_credentials.py)
 - [ImpersonatedCredentials.java](https://github.com/googleapis/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java)

(disclaimer, i'm the author of those two PRs so i may have just gone down this naming path...

for the second part, you mean just go in another name outright like `dataSourceGoogleServiceAccountAccessToken()`
and in config `google_service_account_access_token`? (tht'd also be different then our client library naming (if anything else, atleast with the current name its consistent)

Reply:
Hey @salrashid123, thanks for working through this! I'd like to throw my two cents in here.

I think there's a few important considerations here that weight me towards `google_service_account_access_token` over `google_impersonated_credential`.

* The "domain" for this provider is Google Cloud Platform resources / APIs, so as a baseline we should lean towards GCP concepts- for the provider the part that matters is that we're getting our SA's access token, not that we're performing impersonation or delegation. Meanwhile, on the other hand, an auth library cares more about OAuth as a domain.
* A name like `google_service_account_access_token` feels more declarative than `google_impersonated_credential`; while `impersonated_credential` describes *how* we're getting it, `service_account_access_token` describes *what* we're getting.
* In the provider [`credentials` means a service account key file](https://www.terraform.io/docs/providers/google/provider_reference.html#credentials) while the OAuth token-based auth is named `access_token`, and I value internal provider consistency _much_ more than consistency with external client libraries built for a different domain.

Hope that helps describe some of the reasoning for our end!

Reply:
sure, its fine.  I'll alter the name of the datasource to `google_service_account_access_token`